opba-protocols/xs2a-protocol/src/main/java/de/adorsys/opba/protocol/xs2a/service/xs2a/ais/Xs2aTransactionListingService.java

```diff
@@ -49,30 +51,32 @@ protected void doValidate(DelegateExecution execution, TransactionListXs2aContex
     @SuppressWarnings("checkstyle:MagicNumber") // Hardcoded as it is POC, these should be read from context
     protected void doRealExecution(DelegateExecution execution, TransactionListXs2aContext context) {
         ValidatedPathQueryHeaders<Xs2aResourceParameters, Xs2aTransactionParameters, Xs2aWithConsentIdHeaders> params =
-                extractor.forExecution(context);
+            extractor.forExecution(context);
 
-        Response<TransactionsReport> transactionList = ais.getTransactionList(
+        handler.tryActionOrHandleConsentErrors(execution, eventPublisher, () -> {
```

Comment:
Please extract lambda body
```
   Response<TransactionsReport> transactionList = ais.getTransactionList(
                params.getPath().getResourceId(),
                params.getHeaders().toHeaders(),
                params.getQuery().toParameters()
            );
...
```
into separate private function to increase readability

Reply:
Its 11 lines of code. I think it is still very readable.